src/command_modules/azure-cli-sf/azure/cli/command_modules/sf/_factory.py

```diff
@@ -3,35 +3,22 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+from azure.cli.command_modules.sf.config import SfConfigParser
+from azure.cli.core.util import CLIError
+from azure.servicefabric import ServiceFabricClientAPIs
+from azure.cli.command_modules.sf.cluster_auth import ClientCertAuthentication
+from azure.cli.core.commands.client_factory import configure_common_settings
 
-def cf_sf_client(_):
-    from azure.cli.core.util import CLIError
-    from azure.servicefabric import ServiceFabricClientAPIs
-    from azure.cli.command_modules.sf.custom import (
-        sf_get_cert_info, sf_get_connection_endpoint,
-        sf_get_ca_cert_info, sf_get_verify_setting
-    )
-    from azure.cli.command_modules.sf.cluster_auth import (
-        ClientCertAuthentication
-    )
-    from azure.cli.core.commands.client_factory import (
-        configure_common_settings
-    )
 
-    endpoint = sf_get_connection_endpoint()
+def cf_sf_client(_):
+    sf_config = SfConfigParser()
+    endpoint = sf_config.connection_endpoint()
     if endpoint is None:
```

Comment:
`if not endpoint:`

Unless `endpoint == ''` as any special meaning and should be accepted.